mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,405 @@
+                thresholds = np.append(thresholds, [1.0], axis=0)
+                pr_curve_pandas_df = pd.DataFrame(
+                    {"precision": precision, "recall": recall, "thresholds": thresholds}
+                )
+                self._log_pandas_df_artifact(
+                    artifacts, temp_dir, pr_curve_pandas_df, run_id, "precision_recall_curve",
+                    dataset_name, model,
+                )
+
+                pr_auc = sk_metrics.auc(recall, precision)
+                metrics["precision_recall_auc"] = pr_auc
+
+                def plot_pr_curve():
+                    sk_metrics.PrecisionRecallDisplay(
+                        precision, recall,
+                    ).plot()
+
+                if hasattr(sk_metrics, 'PrecisionRecallDisplay'):
+                    self._log_image_artifact(
+                        artifacts, temp_dir, plot_pr_curve, run_id, "precision_recall_curve", dataset_name,
+                    )
+
+                self._log_image_artifact(
+                    artifacts, temp_dir,
+                    lambda: plot_lift_curve(y, y_probs),
+                    run_id, "lift_curve", dataset_name,
+                )
+
+            def plot_confusion_matrix():
+                sk_metrics.ConfusionMatrixDisplay(confusion_matrix=confusion_matrix).plot()
+
+            if hasattr(sk_metrics, 'ConfusionMatrixDisplay'):
+                self._log_image_artifact(
+                    artifacts, temp_dir, plot_confusion_matrix, run_id, "confusion_matrix", dataset_name,
+                )
+
+        self._log_metrics(run_id, metrics, dataset_name)
+
+        if evaluator_config.get('log_model_explainability', True):
+            self._log_model_explainability(
+                artifacts, temp_dir, model, X, dataset_name, feature_names, run_id, evaluator_config
+            )
+
+        return EvaluationResult(metrics, artifacts)
+
+    def _evaluate_regressor(self, temp_dir, model, X, y, dataset_name, feature_names, run_id, evaluator_config):
+        metrics = EvaluationMetrics()
+        artifacts = {}
+        y_pred = model.predict(X)
+        metrics["example_count"] = len(X)
+        metrics["mean_absolute_error"] = sk_metrics.mean_absolute_error(y, y_pred)
+        metrics["mean_squared_error"] = sk_metrics.mean_squared_error(y, y_pred)
+        metrics["root_mean_squared_error"] = math.sqrt(metrics["mean_squared_error"])
+        metrics['sum_on_label'] = sum(y)
+        metrics['mean_on_label'] = metrics['sum_on_label'] / metrics["example_count"]
+        metrics['r2_score'] = sk_metrics.r2_score(y, y_pred)
+        metrics['max_error'] = sk_metrics.max_error(y, y_pred)
+        metrics['mean_absolute_percentage_error'] = \
+            sk_metrics.mean_absolute_percentage_error(y, y_pred)
+        self._log_model_explainability(
+            artifacts, temp_dir, model, X, dataset_name, feature_names, run_id, evaluator_config
+        )
```

Comment:
Here we're not checking if log model explainability is enabled before calling. Can we pull this call out of `_evaluate_regressor` / `_evaluate_classifier` so that we can check the boolean flag and call it in one place?